### PR TITLE
Enhance/#8143 - Implement the zero data state of an Audience Tile

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #8143 

## Relevant technical choices

* The IB states to add stories for the `AudienceTileCollectingData` and `AudienceTileCollectingDataHideable` components. However, this PR improves by adding stories in the `AudienceTile` and the `AudienceTilesWidget` components. This approach provides better coverage and a more comprehensive view of the components' behavior in various scenarios.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-segmentation-tile.scss [hunk 1]
```diff
@@ -17,6 +17,11 @@
  */
 
 .googlesitekit-plugin {
+	.googlesitekit-widget.googlesitekit-audience-segmentation-tile--zero-data {
+		border-bottom-left-radius: 0;
+		border-bottom-right-radius: 0;
+	}
+
```
Comment (techanvil) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-segmentation-tile.scss:24]: As per my other comment, I don't think we'll want square corners for this version of the widget and we should be able to remove this.

```suggestion
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js [hunk 2]
```diff
@@ -204,6 +213,66 @@ function AudienceTilesWidget( { Widget } ) {
...
+		configuredAudiences.reduce( ( acc, audienceResourceName ) => {
+			acc[ audienceResourceName ] =
+				select( MODULES_ANALYTICS_4 ).isAudiencePartialData(
+					audienceResourceName
+				);
+			return acc;
+		}, {} )
+	);
+
+	const audiencesToReDismiss = [];
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js:239]: I'd suggest renaming this to be a bit more clear about the intent, which is really to clear the dismissal state:

```suggestion
	const audiencesToClearDismissal = [];
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js [hunk 3]
```diff
@@ -204,6 +213,66 @@ function AudienceTilesWidget( { Widget } ) {
...
+				return true;
+			}
+			return true;
+		} )
+		.reverse();
+
+	useEffectOnce( () => {
+		if ( audiencesToReDismiss.length > 0 ) {
+			Promise.all(
+				audiencesToReDismiss.map( ( audienceResourceName ) =>
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js:267]: Along with renaming the variable to e.g. `audiencesToClearDismissal`, I'd suggest adding a comment to explain that we're re-dismissing with a short expiry in order to effectively clear the dismissal state for the item.

**Review by techanvil — CHANGES_REQUESTED** [review]
Thanks @hussain-t, this is looking good! Just a few comments from me, please take a look.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTile/index.js [hunk 4]
```diff
@@ -103,8 +109,43 @@ export default function AudienceTile( {
...
-		return <PreviewBlock width="100%" height="500px" />;
+	if ( ! loaded || isZeroData === undefined || isPartialData === undefined ) {
+		return <PreviewBlock width="100%" height="600px" />;
+	}
+
+	if ( isPartialData && isZeroData ) {
+		return (
+			<Widget
+				noPadding
+				className="googlesitekit-audience-segmentation-tile--zero-data"
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTile/index.js:120]: Thanks for bringing this up, @techanvil. I was wondering about it too, but assumed it was intentional.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTile/AudienceTileCollectingDataHideable.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * Internal dependencies
+ */
+import Link from '../../../../../../components/Link';
+import VisibilityIcon from '../../../../../../../svg/icons/visibility.svg';
+
+export default function AudienceTileCollectingDataHideable( { onHideTile } ) {
+	return (
+		<Fragment>
+			<p className="googlesitekit-audience-segmentation-tile--zero-data__description">
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTile/AudienceTileCollectingDataHideable.js:39]: Thanks for the clarification 👍

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js [hunk 6]
```diff
@@ -204,6 +213,66 @@ function AudienceTilesWidget( { Widget } ) {
...
+				audiencesToReDismiss.push( audienceResourceName );
+				return true;
+			}
+			return true;
+		} )
+		.reverse();
+
+	useEffectOnce( () => {
+		if ( audiencesToReDismiss.length > 0 ) {
+			Promise.all(
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js]: Updated.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js (hunk 3 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js]: Updated.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js (hunk 2 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js]: Updated.

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-segmentation-tile.scss (hunk 1 above)
Comment (hussain-t) [reply assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-segmentation-tile.scss]: Removed.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js [hunk 7]
```diff
@@ -204,6 +213,66 @@ function AudienceTilesWidget( { Widget } ) {
...
+				audienceResourceName
+			);
+
+			if ( isDismissed && isZeroData && configuredAudiences.length > 1 ) {
+				return false;
+			}
+			if ( isDismissed && ! isZeroData ) {
+				// Collect audiences to re-dismiss if they have data again.
+				audiencesToReDismiss.push( audienceResourceName );
+				return true;
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js]: Removed.

**Review by techanvil — CHANGES_REQUESTED** [review]
Thanks @hussain-t, just a last couple of points, one is optional!

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-segmentation-tile.scss [hunk 8]
```diff
@@ -208,5 +208,41 @@
...
+			.googlesitekit-audience-segmentation-tile--zero-data__content {
+				padding: 155px 0;
+			}
+
+			.googlesitekit-audience-segmentation-tile--zero-data__image {
+				margin-bottom: 20px;
+				width: 106px;
+			}
+
+			.googlesitekit-audience-segmentation-tile__description--zero-data {
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-segmentation-tile.scss]: Updated.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js [hunk 9]
```diff
@@ -204,6 +213,65 @@ function AudienceTilesWidget( { Widget } ) {
...
+				report,
+				audienceResourceName
+			);
+
+			if ( isDismissed && isZeroData && configuredAudiences.length > 1 ) {
+				return false;
+			}
+			if ( isDismissed && ! isZeroData ) {
+				// Collect audiences to re-dismiss if they have data again.
+				audiencesToClearDismissal.push( audienceResourceName );
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget.js]: Updated.

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**